### PR TITLE
Enable cluster-sync with external provider

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
Running latest kubemacpool at external cluster like openshift helps on iterating for defects, this PR add support for `KUBEVIRT_PROVIDER=external` and it will deploy and run functest on an external cluster pointed at with `KUBECONFIG`

Doing the following steps will will deploy kubemacpool + kubevirt + multus and wil
run functest towards it at an external cluster.

```
export KUBEVIRT_PROVIDER=external
export KUBECONFIG=[]/kubeconfig
export DEV_REGISTRY=docker.io
export REPO=$USER
export make cluster-up cluster-sync functest
```

**Special notes for your reviewer**:
The installed kubevirt and CNAO/multus is not cleaned up it can be put something in place at a follow up.

**Release note**:

```release-note
Support for KUBEVIRT_PROVIDER=external
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,6 +11,7 @@ export GOFLAGS=-mod=vendor
 export GO111MODULE=on
 export GOROOT=$(BIN_DIR)/go/
 export GOBIN=$(GOROOT)/bin/
+export GOCACHE=$(GOROOT)/cache/
```
Comment (RamLavi) [inline Makefile:14]: what is it used for?

Comment (qinqon) [reply Makefile:14]: Ohh I have to remove this

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # Image URL to use all building/pushing image targets
 REGISTRY ?= quay.io
+REPO ?= kubevirt
```
Comment (ormergi) [inline Makefile:3]: Can you please move Make file changes to another commit?

File: config/test/kustomization.yaml [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
-patchesStrategicMerge:
-- manager_image_patch.yaml
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
+patchesStrategicMerge:
+- manager_image_patch.yaml
+resources:
+- ../default
+images:
+- name: quay.io/kubevirt/kubemacpool
```
Comment (ormergi) [inline config/test/kustomization.yaml:8]: Hey I think we need to take the image from local registry because its the dev environment manifest (config/test)
if we take the image from quay the changes we do and apply with cluster-sync will have no affect.

File: Makefile (hunk 2 above)
Comment (qinqon) [reply Makefile:3]: This bit is needed, the use case for external is something like this 

```bash
export KUBECONFIG=....
export KUBEVIRT_PROVIDER=external
export DEV_REGISTRY=docker.io
export REPO=quiquell
make cluster-sync
```
So we need to parameterize REPO since is going to be the intermediate repository between user and cluster.

File: config/test/kustomization.yaml (hunk 3 above)
Comment (qinqon) [reply config/test/kustomization.yaml:8]: Ups I have break this, let me fix, problem is that I did not fix it well here, that every time you run with external kustomize change this (I have use kustomize edit to change images) maybe we have to use other mechanism.

File: config/test/kustomization.yaml [hunk 4]
```diff
@@ -1,7 +1,9 @@
...
+  newName: docker.io/quiquell/kubemacpool
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply config/test/kustomization.yaml:9]: Oh you keep the old one for release branches?

File: Makefile [hunk 5]
```diff
@@ -41,6 +42,7 @@ generate-deploy: manifests
 	kustomize build config/release > config/release/kubemacpool.yaml
 
 generate-test: manifests
+	cd config/test && kustomize edit set image quay.io/kubevirt/kubemacpool=$(REGISTRY)/$(IMG)
 	kustomize build config/test > config/test/kubemacpool.yaml
```
Comment (RamLavi) [inline Makefile:46]: make sure to change all the kustomize appropriately after #134 merged

File: Makefile (hunk 2 above)
Comment (ormergi) [reply Makefile:3]: I meant to the all the changes in  Makefile so it will be easier to review :)

File: Makefile [hunk 6]
```diff
@@ -59,6 +60,12 @@ generate-deploy: $(KUSTOMIZE) manifests
 generate-test: $(KUSTOMIZE) manifests
 	$(KUSTOMIZE) build config/test > config/test/kubemacpool.yaml
 
+generate-external: $(KUSTOMIZE) manifests
+	cp -r config/test config/external
+	cd config/external && \
```
Comment (ormergi) [inline Makefile:65]: Do we return to the last directory we were?

File: config/test/kubemacpool.yaml [hunk 7]
```diff
@@ -293,7 +293,7 @@ spec:
           value: 0.0.0.0
         - name: HEALTH_PROBE_PORT
           value: "9440"
-        image: registry:5000/kubevirt/kubemacpool:latest
+        image: quay.io/kubevirt/kubemacpool:latest
```
Comment (ormergi) [inline config/test/kubemacpool.yaml:296]: Dont we need to take the image from the local registry container so cluster-sync will affect?

File: cluster/sync.sh [hunk 8]
```diff
@@ -19,25 +19,41 @@ set -ex
 source ./cluster/kubevirtci.sh
 kubevirtci::install
 
-registry_port=$(./cluster/cli.sh ports registry | tr -d '\r')
-registry=localhost:$registry_port
+if [[ "$KUBEVIRT_PROVIDER" == external ]]; then
+    if [[ ! -v DEV_REGISTRY ]]; then
```
Comment (ormergi) [inline cluster/sync.sh:23]: I think `$` is missing before `DEV_REGISTRY`

File: Makefile (hunk 6 above)
Comment (qinqon) [reply Makefile:65]: Yep, shell snippets are Make are very local, in fact they don't change dirs per line that's why we have to do  `&&`

File: config/test/kubemacpool.yaml (hunk 7 above)
Comment (qinqon) [reply config/test/kubemacpool.yaml:296]: This is not to be like this... argg will fix thanks.

File: cluster/sync.sh (hunk 8 above)
Comment (qinqon) [reply cluster/sync.sh:23]: Nope, since it's checking the existence of a var name
